CheckEngine/Emissions/EmissionReport/Categories/Other/Nsps4tSummaryDataCategory.cs

```
@@ -52,46 +52,42 @@ public static bool ExecuteChecks(cSummaryValueInitializationCategory summaryValu
+                    emparams.EmLocationProgramRecords 
+                        = new CheckDataView<VwMpLocationProgramRow>(nsps4tSummaryDataCategory.FilterLocation("LocationProgram", 
+                                                                                                             nsps4tSummaryDataCategory.LocationProgram,
+                                                                                                             emparams.CurrentMonitorPlanLocationRecord.MonLocId));
 
+                    /* Set CurrentMonLocId in the category object */
+                    nsps4tSummaryDataCategory.CurrentMonLocId = emparams.CurrentMonitorPlanLocationRecord.MonLocId;
 
-                        /* Execute the actual checks */
-                        if (!nsps4tSummaryDataCategory.ProcessChecks())
-                        {
-                            result = false;
-                        }
 
-                        /* Erase any parameters created while processing the check category for this location */
-                        nsps4tSummaryDataCategory.EraseParameters();
+                    /* Execute the actual checks */
+                    if (!nsps4tSummaryDataCategory.ProcessChecks())
+                    {
+                        result = false;
                     }
+
+                    /* Erase any parameters created while processing the check category for this location */
+                    nsps4tSummaryDataCategory.EraseParameters();
                 }
-                else
-                {
-                    // Update errors with the returned init check band error.
-                    summaryValueInitializationCategory.Process.UpdateErrors(string.Format("Category: {0}  Message: {1}", nsps4tSummaryDataCategory.CategoryCd, errorMessage));
-                    result = false;
-                }
+            }
+            catch (Exception ex)
```

I assume this exists because the Check Band implementation for NSPS4T differs from the previously implemented Check Categories.

---

This is to account for the fact that `InitCheckBands` now throws an error instead of returning false: all logic that was previously in the `else` block was moved to the `catch` block. The `InitCheckBands` implementation is not different, but the context it's called in is.

However, it's worth noting that the return value isn't used in the one place it's called (in `cEmissionsReportProcess.ExecuteChecksWork`, so may be better to let the error bubble here, too, but I'm not sure that's in the scope of this ticket.